src/formatting/lists.rs

```diff
@@ -444,15 +444,28 @@ where
                 };
                 let width = formatting.shape.width.checked_sub(overhead).unwrap_or(1);
                 let offset = formatting.shape.indent + overhead;
-                let comment_shape = Shape::legacy(width, offset);
+
+                // If `normalize_comments` is not set, second and on one-line comments should
+                // be formatted without the item's line indentation, since the separator
+                // is moved right after the list item, making these (non-first) post-comments
+                // unrelated to the item.
+                let comment_start_trimmed = comment.trim_start();
+                let comment_shape = if !formatting.config.normalize_comments()
+                    && comment_start_trimmed.starts_with("//")
```

Comment:
What's the thinking for scoping this solely to line comments? Won't the same issue described in #3847 exists here too?

e.g. this snippet has the same idempotency issue
```rust
type T1 = Result<
    u32 /* Second comment with some info*/
        /* First longgggggggggggggggggggggggggggggggggg Comment */
        /* First abcd Comment */
        ,
    bool,
>;
```

Reply:
> What's the thinking for scoping this solely to line comments? Won't the same issue described in #3847 exist with block comments too?

Missed that case :disappointed:  Added it now with handling first one-line block comment `/* ... */` the same as one-line open-comment `// ...`.  Also added related test cases.

Reply:
No worries! From a rustfmt POV comments tend to involve myriad edge cases and often end up being a bit annoying :laughing: 

The other case to consider here is when the first line starts a multiline block comment, but there are also additional comments between the end of that block comment and the next list item:

```rust
type T1 = Result<
    u32 /* Second comment with some info 
        * First longgggggggggggggggggggggggggggggggggg Comment */
        /* First abcd Comment */
        ,
    bool,
>;
```

This is starting to feel like a scenario where we may need a more comprehensive picture/processing of the collection of comments within the span, since the typical start/end inspections are not going to be sufficient to cover all cases.

Please also remember that there are various other utilities within the `comments` module that could potentially be helpful (such as `comment_style`). Don't feel obligated to use any of them though, just mentioning it as a reminder in case

Reply:
> The other case to consider here is when the first line starts a multiline block comment, but there are also additional comments between the end of that block comment and the next list item:

Added handling of this case.  It is more complicated, as formatting of the a first multiline comment should use different `shape` than the other comments.

(While evaluating this change I found an issue with formatting multiline comments.  Since this is a general issue I submitted the proposed fix in a separate PR #4617.)

>Please also remember that there are various other utilities within the comments module that could potentially be helpful (such as comment_style). Don't feel obligated to use any of them though, just mentioning it as a reminder in case

Made some changes to use the available utilities when possible.